docs/en/sim_sih/index.md

```diff
@@ -43,20 +43,53 @@ SIH provides several benefits over HITL:
 
 To run the SIH, you will need a:
 
-- [Flight controller](../flight_controller/index.md), such as a Pixhawk-series board
+- [Flight controller](../flight_controller/index.md), such as a Pixhawk-series board.
+
+  ::: info
+  From PX4 v1.14 you can run [SIH "as SITL"](#sih-as-sitl-no-fc), in which case a flight controller is not required.
+  :::
+
 - [Manual controller](../getting_started/px4_basic_concepts.md#manual-control): either a [radio control system](../getting_started/rc_transmitter_receiver.md) or a [joystick](../config/joystick.md).
 - QGroundControl for flying the vehicle via GCS.
 - Development computer for visualizing the virtual vehicle (optional).
 
-From PX4 v1.14 you can run SIH "as SITL", in which case a flight controller is not required.
+## Add SIH to Firmware
+
+The modules required for SIH are not built into all PX4 firmware by default, so you are likely to need to add them to the configuration and then rebuild and install the firmware.
+
+The following keys must be present in the configuration file for your target flight controller (such as [boards/px4/fmu-v6x/default.px4board](https://github.com/PX4/PX4-Autopilot/blob/main/boards/px4/fmu-v6x/default.px4board) FMUv6x based boards).
+
+```text
+CONFIG_MODULES_SIMULATION_PWM_OUT_SIM=y
+CONFIG_MODULES_SIMULATION_SENSOR_BARO_SIM=y
+CONFIG_MODULES_SIMULATION_SENSOR_GPS_SIM=y
+CONFIG_MODULES_SIMULATION_SENSOR_MAG_SIM=y
+```
+
+Add the keys if necessary then re-built the firmware and flash it to the board.
+
+You can alternatively use the following command to launch a GUI configuration tool, and interactively enable them at the path: **modules > Simulation > simulator_sih**.
+For example, to update the fmu-v6x configuration you would use:
+
+```sh
+make px4_fmu-v6x boardconfig
+```
+
+After uploading the following modules should be present on the board: [`pwm_out_sim`](../modules/modules_driver.md#pwm-out-sim), [`sensor_baro_sim`](../modules/modules_system.md#sensor-baro-sim), [`sensor_gps_sim`](../modules/modules_system.md#sensor-gps-sim) and [`sensor_mag_sim`](../modules/modules_system.md#sensor-mag-sim).
+
+You can check that the modules are present using QGroundControl:
+
+1. Open **Analyze Tools > Mavlink Console**.
+2. Type `pwm_out_sim status`, `sensor_baro_sim status`, `sensor_gps_sim status `and `sensor_mag_sim status`, in the console.
+3. If any of the returned values are `nsh: MODULENAME: command not found`, then you don't have the module installed and you need to recompile your target with it.
 
 ## Starting SIH
 
 To set up/start SIH:
 
 1. Connect the flight controller to the desktop computer with a USB cable.
-1. Open QGroundControl and wait for the flight controller too boot and connect.
-1. Open [Vehicle Setup > Airframe](../config/airframe.md) then select the desired frame:
+2. Open QGroundControl and wait for the flight controller too boot and connect.
+3. Open [Vehicle Setup > Airframe](../config/airframe.md) then select the desired frame:
```

Comment:
OK, so are the enabling parameters for these new modules added above in https://github.com/PX4/PX4-Autopilot/pull/25095/files#r2165825880 all enabled? Or should we be doing that here too?

Adding an airframe suggests probably not, where we have to do

```
param set-default SENS_EN_GPSSIM 1
param set-default SENS_EN_BAROSIM 1
param set-default SENS_EN_MAGSIM 1
param set-default SENS_EN_ARSPDSIM 1
```

Reply:
Setting these parameters seems to be needed only for _SIH in SITL_ not for true SIH.
I see them setted only in the _sihsim_ airframes, not in the _sih_ ones

- _sihsim_ airframe: https://github.com/PX4/PX4-Autopilot/blob/main/ROMFS/px4fmu_common/init.d-posix/airframes/10040_sihsim_quadx
- _sih_ airframe: https://github.com/PX4/PX4-Autopilot/blob/main/ROMFS/px4fmu_common/init.d/airframes/1100_rc_quad_x_sih.hil

In true HIL the modules are started in the rcS file:
https://github.com/PX4/PX4-Autopilot/blob/a5a66399745cc7de6f0e55afd08a9c95e46bdfb6/ROMFS/px4fmu_common/init.d/rcS#L317-L333